packages/amplify_api/ios/Classes/FlutterApiErrorUtils.swift

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+import AmplifyPlugins
+
+
+class FlutterApiErrorUtils {
+    
+    static func handleAPIError(flutterResult: @escaping FlutterResult, error: APIError, msg: String, rawResponse: String? = nil) {
+        var errorMap =
+            createErrorMap(localizedError: error.errorDescription, recoverySuggestion: error.recoverySuggestion)
+        if(rawResponse != nil) {
+            errorMap["rawResponse"] = rawResponse
+        }
+        createFlutterError(flutterResult:flutterResult, msg: msg, errorMap: errorMap)
+    }
```

Comment:
This is a field in the details map, so it will present in the ApiError.details in Dart as well when this error occurs.
The `transformationError` case is explicitly handled in iOS, when the response from the server that could not be transformed into the specified response type. The `RawGraphQLResponse` contains the entire response from the service, and so it makes sense to add it to the details. 
In the future if Android handles this case explicitly and provides the `RawGraphQLResponse`, we can add it to the details for Android as well. Also deserialization and transformation is a library level bug, so it may occur in one platform and not the other for a given API call, but I do prefer the way Amplify iOS explicitly handles it and provides the raw response.

Reply:
Gotcha, so for this particular case, the `transformationError` in either platform results in exactly the same error in Dart?